.venv/lib/python3.11/site-packages/PIL/GifImagePlugin.py

```
@@ -0,0 +1,1197 @@
+            elif self.global_palette:
+                from copy import copy
+
+                self.palette = copy(self.global_palette)
+            else:
+                self.palette = None
+        else:
+            if self.mode == "P":
+                if (
+                    LOADING_STRATEGY != LoadingStrategy.RGB_AFTER_DIFFERENT_PALETTE_ONLY
+                    or palette
+                ):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)